### PR TITLE
Ignore last redundant column and null value columns if possible

### PR DESCRIPTION
**Pull request by Linary**

### CODE REVIEW

File: src/test/java/com/baidu/hugegraph/loader/test/functional/LoaderTest.java [hunk 1]
```diff
@@ -536,6 +536,55 @@ public void testLoadVerticesWithJointPrimaryKeys() {
         Assert.assertEquals("Beijing", vertex.property("city"));
     }
 
+    @Test
+    public void testLoadWithIgnoreLastRedudantEmptyColumn() {
```
Comment (zhoney) [inline src/test/java/com/baidu/hugegraph/loader/test/functional/LoaderTest.java]: redundant

File: src/test/java/com/baidu/hugegraph/loader/test/functional/LoaderTest.java [hunk 2]
```diff
@@ -536,6 +536,55 @@ public void testLoadVerticesWithJointPrimaryKeys() {
...
+        // Has a redudant seperator at the end of line
```
Comment (zhoney) [inline src/test/java/com/baidu/hugegraph/loader/test/functional/LoaderTest.java]: ditto

File: src/main/java/com/baidu/hugegraph/loader/parser/ElementParser.java [hunk 3]
```diff
@@ -91,6 +93,19 @@ public GE next() {
         for (String field : this.source().ignoredFields()) {
             keyValues.remove(field);
         }
+
+        SchemaLabel schemaLabel = this.getSchemaLabel(this.source().label());
+        Set<String> nullableKeys = schemaLabel.nullableKeys();
+        Set<Object> nullValues = this.source().nullValues();
+        Iterator<Map.Entry<String, Object>> itor = keyValues.entrySet()
```
Comment (zhoney) [inline src/main/java/com/baidu/hugegraph/loader/parser/ElementParser.java]: test if both nullableKeys and nullValues are not empty first

File: src/main/java/com/baidu/hugegraph/loader/reader/file/TextFileReader.java [hunk 4]
```diff
@@ -81,10 +83,13 @@ protected void initHeader() {
     public Map<String, Object> transform(String line) {
         List<String> columns = this.split(line);
         if (columns.size() != this.header.size()) {
-            throw new ParseException(line,
-                      "The column length '%s' doesn't match with " +
-                      "header length '%s' on: %s",
-                      columns.size(), this.header.size(), line);
+            // Ignore extra separator at the end of the line
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/loader/reader/file/TextFileReader.java]: at the end of line

File: src/main/java/com/baidu/hugegraph/loader/reader/file/TextFileReader.java [hunk 5]
```diff
@@ -96,4 +101,9 @@ protected void initHeader() {
     protected List<String> split(String line) {
         return Splitter.on(this.delimiter).splitToList(line);
     }
+
+    private boolean lastColumnIsEmpty(List<String> columns) {
+        return columns.size() - 1 == this.header.size() &&
+               columns.get(columns.size() - 1).equals(EMPTY_STR);
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/loader/reader/file/TextFileReader.java]: add var: last = columns.size() - 1

File: src/main/java/com/baidu/hugegraph/loader/reader/file/TextFileReader.java [hunk 6]
```diff
@@ -81,10 +83,13 @@ protected void initHeader() {
     public Map<String, Object> transform(String line) {
         List<String> columns = this.split(line);
         if (columns.size() != this.header.size()) {
-            throw new ParseException(line,
-                      "The column length '%s' doesn't match with " +
-                      "header length '%s' on: %s",
-                      columns.size(), this.header.size(), line);
+            // Ignore extra separator at the end of the line
+            if (!this.lastColumnIsEmpty(columns)) {
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/loader/reader/file/TextFileReader.java]: merge to the previous if

File: src/test/resources/struct_null_value.json [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+        "path": "src/test/resources/vertex_person.csv",
+        "format": "CSV",
+        "charset": "UTF-8"
+      },
+      "mapping": {
+        "name": "name",
+        "age": "age",
+        "city": "city"
+      },
+      "null_values": ["NULL", "null", ""]
```
Comment (javeme) [inline src/test/resources/struct_null_value.json]: add to example

**Pull request merged**